app/Providers/AppServiceProvider.php

```diff
@@ -13,7 +13,22 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        //
+        $this->app->singleton(
+            \App\Repositories\Profile\ProfileRepositoryInterface::class,
+            \App\Repositories\Profile\ProfileRepository::class
+        );
+        $this->app->singleton(
+            \App\Repositories\Post\PostRepositoryInterface::class,
+            \App\Repositories\Post\PostRepository::class
+        );
+        $this->app->singleton(
+            \App\Repositories\Like\LikeRepositoryInterface::class,
+            \App\Repositories\Like\LikeRepository::class
+        );
+        $this->app->singleton(
+            \App\Repositories\Follow\FollowRepositoryInterface::class,
+            \App\Repositories\Follow\FollowRepository::class
+        );
```

Comment:
tại sao phải đăng ký ở đây?

Reply:
Theo em hiểu chỗ này thì nó sẽ tạo 1 thực thể từ interface và lấy nó từ class tương ứng. Nếu ko phải thì chị giải thích giúp em, chỗ này em chưa chắc lắm.

Reply:
Cái này đọc dễ hiểu, em xem lại nhé.
https://laravel.com/docs/8.x/container#binding-interfaces-to-implementations
Ngoài ra, singleton khác bind ở đây nhé
https://laravel.com/docs/8.x/container#binding-basics

Reply:
vâng ạ